### PR TITLE
Feature/ticket creation

### PR DESCRIPTION
**Pull request by simpleman19**
Everything should be working, however I can't add it to the database.  @epielemeier Lines 394 and 395 of TicketCreationJPanel are doing the database save, I get an unsaved transient error.  If I change company in Ticket to a transient then everything seems to work just fine.

### CODE REVIEW

**Comment by epielemeier** [comment]
@simpleman19 Try `script/destroy_db.bat` and `script/init_db.bat` and let me know if that works

File: src/acme/pd/Ticket.java [hunk 1]
```diff
@@ -69,56 +66,85 @@
...
+        this.setCompany(company);
+        HashMap<UUID, Customer> customers = new HashMap<>(company.getCustomers());
+        // Set the pickup customer to a customer (Useful for UI)
+        Customer tmpCust = (Customer) customers.values().toArray()[0];
+        this.setPickupCustomer(tmpCust);
+        // Remove customer from map to be sure that pickup and destination are different
+        customers.remove(tmpCust.getId());
+        // Set destination customer to a customer (Useful for UI)
+        this.setDeliveryCustomer((Customer) customers.values().toArray()[0]);
+        this.setDeliveryTime(LocalDateTime.now().plusHours(6));
```
Comment (epielemeier) [inline src/acme/pd/Ticket.java]: `this.setDeliveryTime()` should be handled by the code @dee-dee-raye already wrote for us.

File: src/acme/pd/Ticket.java [hunk 2]
```diff
@@ -215,18 +253,16 @@ public LocalDateTime getDeliveryTime() {
     }
 
     public void setDeliveryTime(LocalDateTime deliveryTime) {
-//        if (new Random().nextInt() % 3 == 0)
-//            //this.bonus = company.getBonus();
-//        else
-//            this.bonus = new BigDecimal(0);
+        // TODO Bonus
```
Comment (epielemeier) [inline src/acme/pd/Ticket.java]: Calculating bonuses should be covered by this user story.

File: test/acme/data/HibernateAdapterTest.java [hunk 3]
```diff
@@ -115,7 +115,7 @@ public void testCustomer() {
 	
 	@Test
 	public void testTicket() {
-		Ticket ticket = new Ticket();
+		Ticket ticket = new Ticket(null);
```
Comment (epielemeier) [inline test/acme/data/HibernateAdapterTest.java]: This looks bad. The default constructor for `Ticket` should have a call to `this(null)` if necessary.

**Comment by epielemeier** [comment]
I added some commits to fix runtime errors on database persistence. The issue was that there were a mixture of persisted and not persisted entities. I added changes so that you can save a ticket to the database on clicking `Save` button.

File: .idea/compiler.xml [hunk 4]
```diff
@@ -4,6 +4,8 @@
     <bytecodeTargetLevel>
       <module name="acmeDSTS_main" target="1.8" />
       <module name="acmeDSTS_test" target="1.8" />
+      <module name="Documents-acmeDSTS_main" target="1.8" />
+      <module name="Documents-acmeDSTS_test" target="1.8" />
     </bytecodeTargetLevel>
   </component>
 </project>
```
[the remark this answers is not in the document]
Comment (simpleman19) [reply .idea/compiler.xml]: yes

[the remark this answers is not in the document]
Comment (simpleman19) [reply .idea/compiler.xml]: The machine specific stuff is ignored but several of the project files are meant to be shared

File: src/acme/pd/Company.java [hunk 5]
```diff
@@ -53,57 +55,87 @@
     private double courierMilesPerHour = 5.8;
     @Transient
     private String mapFile = "map/map.csv";
+    @Transient
+    public final DateTimeFormatter acmeDF = DateTimeFormatter.ofPattern("yyyy-MM-dd hh:mm:ss a");
```
[the remark this answers is not in the document]
Comment (simpleman19) [reply src/acme/pd/Company.java]: Ya, I was just trying to not have to type that many characters to use it, even with it as acmeDF, i was running into my preferred line length limit

File: src/acme/pd/Ticket.java [hunk 6]
```diff
@@ -69,56 +66,85 @@
...
-      
-      this.calcEstimatedTimes();
-        return new BigDecimal(quote);
+    private void updatePath() {
+        // Only update if pickup and delivery are set
+        if (this.getPickupCustomer() != null
+                && this.getDeliveryCustomer() != null
+                && this.getDeliveryTime() != null) {
+            this.path = company.getMap().getPath(
+                    this.getPickupCustomerLocation(),
```
[the remark this answers is not in the document]
Comment (simpleman19) [reply src/acme/pd/Ticket.java]: Why in the world did we make that a requirement?

[the remark this answers is not in the document]
Comment (simpleman19) [reply src/acme/pd/Ticket.java]: I am prettu sure Dee Dee didn't account for that, so we may not get that functionality

File: test/acme/data/HibernateAdapterTest.java (hunk 3 above)
Comment (simpleman19) [reply test/acme/data/HibernateAdapterTest.java]: Can't because of hibernate

File: src/acme/pd/Ticket.java (hunk 2 above)
Comment (simpleman19) [reply src/acme/pd/Ticket.java]: Nope, completion

File: src/acme/pd/Ticket.java (hunk 1 above)
Comment (simpleman19) [reply src/acme/pd/Ticket.java]: She says that she didnt

**Pull request merged**